.github/workflows/func.yaml

```diff
@@ -0,0 +1,32 @@
+name: Test suite
+
+on:
+  pull_request_review:
+    types: [ submitted ]
+
+jobs:
+  func:
+    name: Functional tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    if: >-
+      github.event.review.state == 'approved'
+      || github.event.review.body == 'recheck'
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: lxd
+      - name: Run tests on Bionic series
```

Comment:
Is it possible to use the [matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) solution in [lint-unit.yaml](https://github.com/canonical/charm-simple-streams/blob/main/.github/workflows/lint-unit.yaml) ?
```
    strategy:
      matrix:
        python-version: ["3.6", "3.8", "3.10"]
```
something like this

```
    ubuntu-series:[ "bionic", "focal", "jammy"] 
```

Reply:
> Is it possible to use the [matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) solution in [lint-unit.yaml](https://github.com/canonical/charm-simple-streams/blob/main/.github/workflows/lint-unit.yaml) ?
> 
> ```
>     strategy:
>       matrix:
>         python-version: ["3.6", "3.8", "3.10"]
> ```

You mean running functests with different python version?

> 
> something like this
> 
> ```
>     ubuntu-series:[ "bionic", "focal", "jammy"] 
> ```

It's possible, but I decided not to do it just for the sake of resources. Approach like this will create 3 machines and bootstrap 3 different controllers, what is no needed. Or do you think we should do this?

Reply:
If we can afford it (resource is enough), I prefer to run it parallel to shorten the functional testing time.

Reply:
I think that we have plenty of resources, so we can do parallel. However right now there are only basic func test, which is deploying one unit and we will save less than 5min with this. 

I will rework it to parallel as it will be beneficial in the future.